### DISCUSSION TITLE
Does having too many user filters affect performance?

### DISCUSSION THREAD

**Discussion by ghost** [uBlockOrigin/uAssets#23410] [Q&A] [answered]
I have a custom user filter that I have stored on Github. My filter contains a total of 650 rules, but I decided to remove my filter from Github and add it to uBo's user filter. I can make rule corrections and rule additions faster in uBO.

I'm wondering, will adding too many user filters to the extension reduce the performance of the extension? Like increase RAM usage, slow startup, etc.

**Comment by Yuki2718** [comment]
RAM => yes, filters must be stored somewhere
startup => yes, uBO must initialize and load filters
any other performance metrics such as page load => no, as long as filters are well-written
But even for the first two items, if you measure actual usage, it's usually not something too worry about. Always best to rigrously benchmark.

**Reply by garry-ut99** [reply]
OP's question has a context highlighted at the beginning of his comment, hence it seems he more likely was asking for a difference after moving his filter list from external URL into "My filters":

- there is no performance or RAM difference when using a filter list from outside of uBO compared to having the same list copy & pasted into "My filters", because technically "My filters" is just same filter list like every other filter lists: regardless of where does a filter list comes from: an external URL or pasted into "My filters", the filters are being processed the same way.

<hr>

And as for answering the question without that context, several hundred or thousand filters doesn't make any noticeable difference compared to default uBO filter lists which already contain over 181000 filters (as long as OP doesn't use many regex filters.).

Also more general information about resource usage of uBO as a whole is available in documentation:
https://github.com/gorhill/uBlock/wiki in section "uBlock's resource usage and efficiency"

**Comment by ghost** [comment]
> OP's question has a context highlighted at the beginning of his comment, hence it seems he more likely was asking for a difference after moving his filter list from external URL into "My filters":

Yeah, that's what I was talking about. Thanks for your explanation.

------------------

And I have one more question.


How can I import a filter of 8 text files using a single link?

I mean, for example, the EasyList filter is actually 16 different text files, but uBlock Origin merges the EasyList into one piece. 
Like this link: https://ublockorigin.github.io/uAssets/thirdparties/easylist.txt
https://i.ibb.co/s1NMvM0/easypng.png

This link contains the whole EasyList and is very practical.

I couldn't find how to do it, how can I import my filter into uBO as a single piece like EasyList?

My native language is not English, I hope you understand what I say.

**Reply by ghost** [reply, marked as answer]
Lists can be merged server side using templates and scripts like these https://github.com/uBlockOrigin/uAssets/blob/master/templates/easylist.template https://github.com/uBlockOrigin/uAssets/blob/master/tools/make-easylist.mjs

or by uBO `!#include` directive https://github.com/uBlockOrigin/uBlock-issues/wiki/Static-filter-syntax#include-file-name

**Reply by garry-ut99** [reply]
Also here is some novice user friendly example / discussion about using `!include` directive :
- https://github.com/yourduskquibbles/webannoyances/issues/67
- https://github.com/yourduskquibbles/webannoyances/blob/master/ultralist.txt

**Comment by ghost** [comment]
Not completely related to the discussion, but with the following rules applied on YT, in a live stream with chat running at 2 to 3 messages/sec, profile pictures of the accounts are not loaded (which I intend to) but within few minutes of the live stream playing the tab crashes completely. So very frequent network request cancellations do affect performance sometimes for me.  
```
* * 3p block
* * 3p-frame block
* * 3p-script block
www.youtube.com googlevideo.com * noop
www.youtube.com ytimg.com * noop
```

**Reply by garry-ut99** [reply]
Report crashes to browser devs, uBO is not responsible for crashes, see: https://github.com/gorhill/uBlock/wiki/Can-uBO-crash-a-browser%3F
